jstz-dev/jstz
# feat(octez): implement execute_rollup_outbox_message

# Context

Completes JSTZ-145.

[JSTZ-145](https://linear.app/tezos/issue/JSTZ-145/implement-octezclientexecute-rollup-outbox-message)

# Description

Implement `execute_rollup_outbox_message` that performs `octez-client execute outbox message of smart rollup ...`.

# Manually testing the PR

* Integration test: added one test. The setup is as follows:
  * contract: takes input `int(x)` and does something.
  * rollup: writes a message `int(1)` targeting the contract to the outbox whenever an external message arrives at the inbox. Source file is included in this PR as well. Since there is no one else writing messages to the inbox and there is no other rollup running, it's safe to assume that the only outbox message is produced by this exact rollup based on the inbox messages sent by the only call to `octez-client write inbox message` in the test.

  The addresses and contract files are brand new and separated from existing parameter files in order to prevent unexpected changes to the contract files that affect the tests. Housekeeping will be done in a separate PR to organise other test files and keep the base test parameter template clean, i.e. bootstrap contracts and rollups should be injected with `ParameterBuilder` rather than hardcoded in the template.
